public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -15,6 +16,21 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptio
     [query.azureMonitor?.dimensionFilters]
   );
 
+  const dimensionOperators: Array<SelectableValue<string>> = [
+    { label: '==', value: 'eq' },
+    { label: '!=', value: 'ne' },
+    { label: 'starts with', value: 'sw' },
+  ];
+
+  const validDimensionOptions = useMemo(() => {
+    let t = dimensionOptions;
+    let dimensionFilters = query.azureMonitor?.dimensionFilters;
+    if (dimensionFilters !== undefined && dimensionFilters.length > 0) {
+      t = dimensionOptions.filter((val) => !dimensionFilters?.find((dimension) => dimension.dimension === val.value));
```

Comment:
Maybe add a comment? This bit of logic is a little complex

Reply:
A few nits: 
An undefined dimensionFilters and an empty array both evaluates to false, so you should be able to replace the if condition to `if (query.azureMonitor?.dimensionFilters?.length) {`. 

Think you can also remove the not-null assertion operator from the filter condition. Using the `dimensionFilters?.find` is fine, but using `dimensionFilters?.some` would be a bit more expressive in this case. 

 Maybe a unit test can be added for this?